cibyl/outputs/cli/ci/system/common/builds.py

```
@@ -0,0 +1,94 @@
+    :rtype: bool
+    """
+    return build.status.value
+
+
+def get_status_section(palette, build):
+    """Generates the text describing the status of a build.
+
+    :param palette: The palette of colors to follow.
+    :type palette: :class:`cibyl.utils.colors.ColorPalette`
+    :param build: The build to get the data from.
+    :type build: :class:`cibyl.models.ci.base.build.Build`
+    :return: The text with the status of the build.
+    :rtype: str
+    """
+    if not has_status_section(build):
+        raise ValueError('Build has no status to create a section for.')
+
+    text = IndentedTextBuilder()
+
+    status_x_color_map = {
+        'SUCCESS': lambda: palette.green(build.status.value),
+        'FAILURE': lambda: palette.red(build.status.value),
+        'UNSTABLE': lambda: palette.yellow(build.status.value)
+    }
+
+    status = status_x_color_map.get(
+        build.status.value,
+        lambda: palette.underline(build.status.value)
+    )()
+
+    text.add(palette.blue('Status: '), 0)
+    text[-1].append(status)
+
+    return text.build()
+
+
+def has_duration_section(build):
+    """Checks whether a build has enough data to build a duration entry for
+    its description.
+
+    :param build: The build to check.
+    :type build: :class:`cibyl.models.ci.base.build.Build`
+    :return: True if a duration section can be built, False if not.
+    :rtype: bool
+    """
+    return build.duration.value
+
+
+def get_duration_section(palette, build):
+    """Generates the text describing the duration of a build.
+
+    :param palette: The palette of colors to follow.
+    :type palette: :class:`cibyl.utils.colors.ColorPalette`
+    :param build: The build to get the data from.
+    :type build: :class:`cibyl.models.ci.base.build.Build`
+    :return: The text with the duration of the build.
+    :rtype: str
+    """
+    if not has_duration_section(build):
+        raise ValueError('Build has no duration to create a section for.')
```

Do we want to raise an exception here? I would think that just returning early without printing anything would be sufficient. For example I think that we saw that some builds had a malformed duration field in elasticsearch sometimes, and we left that field empty. Also, I have not seen this exception explicitly caught anywhere, so if it gets raised it will not follow the same output as the CibylException types.